06_puppy/alextmz/mapstore.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"errors"
+)
+
+type MapStore map[uint16]*Puppy
+
+func NewmapStore() *MapStore {
+	a := MapStore{}
+	return &a
+}
+
+func (m *MapStore) CreatePuppy(p *Puppy) error {
+	if _, ok := (*m)[p.ID]; !ok {
+		(*m)[p.ID] = p
+		return nil
+	}
+	return errors.New("puppy ID being created already exists")
+}
+
+func (m *MapStore) ReadPuppy(id uint16) (*Puppy, error) {
+	if v, ok := (*m)[id]; ok {
+		return v, nil
+	}
+	return nil, errors.New("puppy ID being read does not exist")
```

Comment:
would be good to indicate the id being read in the error e.g  puppy with id: %d"

Reply:
Yes. Will check that.